### PR TITLE
Upgrade vmactions to use linux runners

### PR DESCRIPTION
**Pull request by Neilpang**
Upgrade the vmactions to the latest vesions to use the linux runners.

### CODE REVIEW

**Comment by azat** [comment]
This is just wonderful, let's see on CI!

**Review by azat — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -669,14 +669,12 @@ jobs:
           path: build
 
   freebsd-cmake-job:
-    runs-on: macos-12
-    if: "false && !contains(github.event.head_commit.message, 'ci/freebsd skip') && !contains(github.event.head_commit.message, 'ci/freebsd/cmake skip')"
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'ci/freebsd skip') && !contains(github.event.head_commit.message, 'ci/freebsd/cmake skip')"
```
Comment (azat) [inline .github/workflows/build.yml:673]: You forgot to enable them, I've fixed it (but with force push, to squash the commits, hope it is OK)

**Comment by azat** [comment]
It does not work, @Neilpang can you please take a look?

**Comment by Neilpang** [comment]
it seems that freebsd 12.4 is not supported anymore,  please just remove 12.4 from the build.

**Review by azat — APPROVED** [review]

**Pull request merged**